### PR TITLE
Ft person profile

### PR DESCRIPTION
**Pull request by iamcristos**
#### What does this PR do?
- creates Person Profile Page

#### Description of Task to be completed?
- create profile component
- create person profile route
- create error toastify message
- styled person profile for desktop view

#### How should this be manually tested?
- clone the repo
- npm install
- npm start
- go to `http://localhost:3000/profile`

#### What are the relevant Trello Card stories? [ Trello Card Item ](https://trello.com/b/8x3mo2V5/say-their-names-web)
- Person Profile Component

#### Pull Request Type (check one)
- [x] Feature

#### Screenshots (if appropriate)
Screenshot 2020-06-02 at 2 07 41 PM

### CODE REVIEW

File: src/pages/PersonProfile.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+const data = [{
+  image: 'https://via.placeholder.com/150',
+  fullname: 'James Bond',
+  age: '42',
+  location: 'SOMEWHERE',
+  context: 'smkjjyhhbbfnfuhhb lorem',
+  children: 2
+}];
+export default function PersonProfile() {
```
Comment (Bedrock02) [inline src/pages/PersonProfile.js:16]: Let's avoid using export default function [component]
See comment above and changes here https://github.com/Say-Their-Name/web/pull/13

File: src/pages/PersonProfile.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+//   const [persons, setPerson] = useState([]);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    async function fetchdata() {
+      try {
+        await axios.get('thebackendurl.com');
+        // setPerson(response);
```
Comment (Bedrock02) [inline src/pages/PersonProfile.js:24]: remove the comment if no longer needed

**Comment by Bedrock02** [comment]
I hope the comments helped, feel free to hit me up if you have any questions. You can also run `yarn lint` to see what linting rules need fixing or I think you can use `yarn lint:fix` ( I would review the changes that yarn runs after this comment)

File: src/components/profile/index.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        <h2>Context</h2>
+        <p>{context}</p>
+      </Context>
+    </Profile>
+  );
+}
+
+index.propTypes = {
+  info: PropTypes.shape({
+    image: PropTypes.string.isRequired,
```
[the remark this answers is not in the document]
Comment (iamcristos) [reply src/components/profile/index.js:63]: eslint automatically lint's before we pus

File: src/pages/PersonProfile.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+const data = [{
+  image: 'https://via.placeholder.com/150',
+  fullname: 'James Bond',
+  age: '42',
+  location: 'SOMEWHERE',
+  context: 'smkjjyhhbbfnfuhhb lorem',
+  children: 2
+}];
+export default function PersonProfile() {
+//   const [persons, setPerson] = useState([]);
```
[the remark this answers is not in the document]
Comment (iamcristos) [reply src/pages/PersonProfile.js:17]: the comment will still be needed once we get a valid API to fetch the data from the backend/db.... then we replace the mock data with the data in state

File: src/pages/PersonProfile.js (hunk 1 above)
Comment (iamcristos) [reply src/pages/PersonProfile.js:16]: aright

File: src/components/profile/index.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import {
+  Profile, PersonalInformation, Photo, PersonSection, Name, H4, Div, Age, Children, Location, ShareSection, Button, H2, Share, Context
+} from './styles';
+
+export default function index({ info }) {
```
[the remark this answers is not in the document]
Comment (iamcristos) [reply src/components/profile/index.js:8]: aright

File: src/components/profile/Button/Button.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+
+import BackProfile from './styles';
+
+function Button() {
```
Comment (Bedrock02) [inline src/components/profile/Button/Button.js:5]: Let's utilize ES6 syntax

```
const Button = () => (
    <BackProfile>
      <button type="button">BACK TO PROFILES</button>
    </BackProfile>
);
```

File: src/components/profile/PersonProfile.js [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+  Div,
+  Age,
+  Children,
+  Location,
+  ShareSection,
+  Button,
+  H2,
+  Share,
+  Context
+} from './styles';
```
Comment (Bedrock02) [inline src/components/profile/PersonProfile.js:20]: 👍

File: src/components/profile/PersonProfile.js [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+
+function PersonProfile({ info }) {
```
Comment (Bedrock02) [inline src/components/profile/PersonProfile.js:22]: Same as above lets use ES6 syntax

```
const PersonProfile = () => {
 .....
 ......
 <VARIABLES HERE>
 return (
   <JSX HERE>
 );

}
```

File: src/components/profile/PersonProfile.js [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+
+PersonProfile.propTypes = {
+  info: PropTypes.shape({
+    image: PropTypes.string.isRequired,
+    fullname: PropTypes.string.isRequired,
+    age: PropTypes.number.isRequired,
+    location: PropTypes.string.isRequired,
+    context: PropTypes.string.isRequired,
+    children: PropTypes.number
+  }).isRequired
```
Comment (Bedrock02) [inline src/components/profile/PersonProfile.js:85]: 👍

File: src/pages/PersonProfile.js [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+
+const data = [{
+  image: 'https://via.placeholder.com/150',
+  fullname: 'James Bond',
+  age: '42',
+  location: 'SOMEWHERE',
+  context: 'smkjjyhhbbfnfuhhb lorem',
+  children: 2
+}];
+export default function PersonProfile() {
```
Comment (Bedrock02) [inline src/pages/PersonProfile.js:16]: ES6 syntax for functions
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions
```
const PersonProfile = () => {
   ......
}

export default PersonProfile;
```

**Comment by Easybuoy** [comment]
> I ran `yarn lint` and I am seeing some lint errors on some of the files that have changes, if we could fix those then we are set.
> 
> ```
> /Users/steven/personal/STN/web/src/components/profile/Button/Button.js
>    5:22  error  Unexpected block statement surrounding arrow body; move the returned value immediately after the `=>`  arrow-body-style
>   11:2   error  Missing semicolon
> 
> /Users/steven/personal/STN/web/src/components/profile/PersonProfile.js
>   73:2  error  Missing semicolon  semi
> 
> 
> /Users/steven/personal/STN/web/src/pages/PersonProfile.js
>   41:2   error  Missing semicolon                              semi
>   43:29  error  Newline required at end of file but not found  eol-last
>   43:29  error  Missing semicolon
> ```

In the husky setup, should we be running `yarn lint:fix` instead of `yarn lint` so some of the 

> I ran `yarn lint` and I am seeing some lint errors on some of the files that have changes, if we could fix those then we are set.
> 
> ```
> /Users/steven/personal/STN/web/src/components/profile/Button/Button.js
>    5:22  error  Unexpected block statement surrounding arrow body; move the returned value immediately after the `=>`  arrow-body-style
>   11:2   error  Missing semicolon
> 
> /Users/steven/personal/STN/web/src/components/profile/PersonProfile.js
>   73:2  error  Missing semicolon  semi
> 
> 
> /Users/steven/personal/STN/web/src/pages/PersonProfile.js
>   41:2   error  Missing semicolon                              semi
>   43:29  error  Newline required at end of file but not found  eol-last
>   43:29  error  Missing semicolon
> ```

This was caused by another pull request that was merged, we need to setup CI for linting I guess

**Comment by Bedrock02** [comment]
Approved! Looks good, squash and merge.

**Pull request merged**